floris/tools/optimization/yaw_optimization/yaw_optimizer_scipy.py

```diff
@@ -83,68 +83,72 @@ def optimize(self):
         """
-        # Loop through every WD and WS individually
+        # Loop through every wind condition individually
         wd_array = self.fi_subset.floris.flow_field.wind_directions
         ws_array = self.fi_subset.floris.flow_field.wind_speeds
-        for nwsi, ws in enumerate(ws_array):
-
-            self.fi_subset.reinitialize(wind_speeds=[ws])
-
-            for nwdi, wd in enumerate(wd_array):
-                # Find turbines to optimize
-                turbs_to_opt = self._turbs_to_opt_subset[nwdi, nwsi, :]
-                if not any(turbs_to_opt):
-                    continue  # Nothing to do here: no turbines to optimize
-
-                # Extract current optimization problem variables (normalized)
-                yaw_lb = self._minimum_yaw_angle_subset_norm[nwdi, nwsi, turbs_to_opt]
-                yaw_ub = self._maximum_yaw_angle_subset_norm[nwdi, nwsi, turbs_to_opt]
-                bnds = [(a, b) for a, b in zip(yaw_lb, yaw_ub)]
-                x0 = self._x0_subset_norm[nwdi, nwsi, turbs_to_opt]
-
-                J0 = self._farm_power_baseline_subset[nwdi, nwsi]
-                yaw_template = self._yaw_angles_template_subset[nwdi, nwsi, :]
-                turbine_weights = self._turbine_weights_subset[nwdi, nwsi, :]
-                yaw_template = np.tile(yaw_template, (1, 1, 1))
-                turbine_weights = np.tile(turbine_weights, (1, 1, 1))
-
-                # Handle heterogeneous inflow, if there is one
-                if (hasattr(self.fi.floris.flow_field, 'heterogenous_inflow_config') and
-                    self.fi.floris.flow_field.heterogenous_inflow_config is not None):
-                    het_sm_orig = np.array(
-                        self.fi.floris.flow_field.heterogenous_inflow_config['speed_multipliers']
-                    )
-                    het_sm = het_sm_orig[nwdi,:].reshape(1,-1)
-                else:
-                    het_sm = None
-
-                # Define cost function
-                def cost(x):
-                    x_full = np.array(yaw_template, copy=True)
-                    x_full[0, 0, turbs_to_opt] = x * self._normalization_length
-                    return (
-                        - 1.0 * self._calculate_farm_power(
-                            yaw_angles=x_full,
-                            wd_array=[wd],
-                            turbine_weights=turbine_weights,
-                            heterogeneous_speed_multipliers=het_sm
-                        )[0, 0] / J0
-                    )
-
-                # Perform optimization
-                residual_plant = minimize(
-                    fun=cost,
-                    x0=x0,
-                    bounds=bnds,
-                    method=self.opt_method,
-                    options=self.opt_options,
-                )
+        for i, wc in enumerate(zip(wd_array, ws_array)):
+
+            wd = wc[0]
+            ws = wc[1]
```

Comment:
I think this can be shortened to:

for i, (wd, ws) in enumerate(zip(wd_array, ws_array)):